File: Test12/step_definitions/Shopping Cart and Checkout.js
import { Given, When, Then } from "cypress-cucumber-preprocessor/steps";
import ShoppingCartPage from "../pages/shoppingCartPage";
import CheckoutPage from "../pages/checkoutPage";

Given("the user has items in the shopping cart", () => {
 ShoppingCartPage.verifyItemsInCart();
});

When("the user adjusts the quantity of a product", () => {
 ShoppingCartPage.adjustProductQuantity();
});

Then("the updated quantity should be reflected in the cart", () => {
 ShoppingCartPage.verifyUpdatedQuantity();
});

Given("the user proceeds to checkout", () => {
 CheckoutPage.proceedToCheckout();
});

When("the user provides valid shipping and billing information", () => {
 CheckoutPage.enterShippingAndBillingInfo();
});

Then("the information should be accepted and allow the user to continue", () => {
 CheckoutPage.verifyInfoAccepted();
});

Given("the user is at the payment options stage of checkout", () => {
 CheckoutPage.verifyAtPaymentOptions();
});

When("the user selects a payment method between credit/debit card and PayPal", () => {
 CheckoutPage.selectPaymentMethod();
});

Then("the selected payment method should be processed correctly", () => {
 CheckoutPage.verifyPaymentProcessed();
});

Given("the user has completed the checkout process", () => {
 CheckoutPage.verifyCheckoutComplete();
});

When("the order is confirmed", () => {
 CheckoutPage.confirmOrder();
});

Then("the user should receive an order confirmation via email and/or SMS", () => {
 CheckoutPage.verifyOrderConfirmation();
});

Given("the user enters payment information", () => {
 CheckoutPage.enterPaymentInformation();
});

When("the payment is processed", () => {
 CheckoutPage.processPayment();
});

Then("the payment data should be securely transmitted and stored", () => {
 CheckoutPage.verifyPaymentSecurity();
});

Given("the user encounters an error during checkout", () => {
 CheckoutPage.simulateError();
});

When("the error occurs", () => {
 CheckoutPage.triggerError();
});

Then("an appropriate error message should be displayed to the user", () => {
 CheckoutPage.verifyErrorMessage();
});

Given("the user completes a checkout", () => {
 CheckoutPage.completeCheckout();
});

When("the checkout process is finalized", () => {
 CheckoutPage.finalizeCheckout();
});

Then("the system should log the checkout activities", () => {
 CheckoutPage.verifyCheckoutLogging();
});

Given("the user selects a payment method", () => {
 CheckoutPage.selectPayment();
});

When("the payment is processed through a gateway", () => {
 CheckoutPage.processPaymentThroughGateway();
});

Then("the integration with the payment gateway should be seamless", () => {
 CheckoutPage.verifyGatewayIntegration();
});

Given("there are changes to the user's checkout status", () => {
 CheckoutPage.simulateCheckoutStatusChange();
});

When("a change occurs", () => {
 CheckoutPage.triggerStatusChange();
});

Then("the user should be notified of the changes", () => {
 CheckoutPage.verifyUserNotification();
});

Given("the user proceeds through the checkout process", () => {
 CheckoutPage.proceedThroughCheckout();
});

When("they interact with all features", () => {
 CheckoutPage.interactWithAllFeatures();
});

Then("each feature should function correctly and reliably", () => {
 CheckoutPage.verifyAllFeaturesFunctionality();
});

File: Test12/step_definitions/Shopping Cart and Checkout.java
package com.cucumber.steps;

import com.page_objects.CartPage;
import com.page_objects.CheckoutPage;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

public class ShoppingCartCheckoutSteps extends CartPage {

    @Given("the user has items in the shopping cart")
    public void theUserHasItemsInTheShoppingCart() {
        addProductToCart();
        openCart();
    }

    @When("the user adjusts the quantity of a product")
    public void theUserAdjustsTheQuantityOfAProduct() {
        updateProductQuantity("2");
    }

    @Then("the updated quantity should be reflected in the cart")
    public void theUpdatedQuantityShouldBeReflectedInTheCart() {
        verifyProductQuantity("2");
    }

    @Given("the user proceeds to checkout")
    public void theUserProceedsToCheckout() {
        proceedToCheckout();
    }

    @When("the user provides valid shipping and billing information")
    public void theUserProvidesValidShippingAndBillingInformation() {
        enterShippingDetails("John Doe", "123 Main St", "Anytown", "12345", "555-1234");
    }

    @Then("the information should be accepted and allow the user to continue")
    public void theInformationShouldBeAcceptedAndAllowTheUserToContinue() {
        // Assume continuation is verified by proceeding to payment
        selectPaymentMethod("Credit Card");
    }

    @Given("the user is at the payment options stage of checkout")
    public void theUserIsAtThePaymentOptionsStageOfCheckout() {
        proceedToCheckout();
    }

    @When("the user selects a payment method between credit/debit card and PayPal")
    public void theUserSelectsAPaymentMethodBetweenCreditDebitCardAndPayPal() {
        selectPaymentMethod("PayPal");
    }

    @Then("the selected payment method should be processed correctly")
    public void theSelectedPaymentMethodShouldBeProcessedCorrectly() {
        placeOrder();
    }

    @Given("the user has completed the checkout process")
    public void theUserHasCompletedTheCheckoutProcess() {
        placeOrder();
    }

    @When("the order is confirmed")
    public void theOrderIsConfirmed() {
        verifyOrderConfirmation("Order confirmed");
    }

    @Then("the user should receive an order confirmation via email and/or SMS")
    public void theUserShouldReceiveAnOrderConfirmationViaEmailAndOrSMS() {
        // Assume confirmation is verified by checking a message
        verifyOrderConfirmation("Confirmation sent via email/SMS");
    }

    @Given("the user enters payment information")
    public void theUserEntersPaymentInformation() {
        selectPaymentMethod("Credit Card");
    }

    @When("the payment is processed")
    public void thePaymentIsProcessed() {
        placeOrder();
    }

    @Then("the payment data should be securely transmitted and stored")
    public void thePaymentDataShouldBeSecurelyTransmittedAndStored() {
        // Assume security is verified by checking a secure transmission message
        verifyOrderConfirmation("Payment data secured");
    }

    @Given("the user encounters an error during checkout")
    public void theUserEncountersAnErrorDuringCheckout() {
        // Simulate an error
        throw new RuntimeException("Checkout error");
    }

    @When("the error occurs")
    public void theErrorOccurs() {
        // Error handling logic
    }

    @Then("an appropriate error message should be displayed to the user")
    public void anAppropriateErrorMessageShouldBeDisplayedToTheUser() {
        // Assume error message is verified
        verifyOrderConfirmation("Error message displayed");
    }

    @Given("the user completes a checkout")
    public void theUserCompletesACheckout() {
        placeOrder();
    }

    @When("the checkout process is finalized")
    public void theCheckoutProcessIsFinalized() {
        // Finalization logic
    }

    @Then("the system should log the checkout activities")
    public void theSystemShouldLogTheCheckoutActivities() {
        // Assume logging is verified
        verifyOrderConfirmation("Checkout activities logged");
    }

    @Given("the user selects a payment method")
    public void theUserSelectsAPaymentMethod() {
        selectPaymentMethod("Credit Card");
    }

    @When("the payment is processed through a gateway")
    public void thePaymentIsProcessedThroughAGateway() {
        placeOrder();
    }

    @Then("the integration with the payment gateway should be seamless")
    public void theIntegrationWithThePaymentGatewayShouldBeSeamless() {
        // Assume seamless integration is verified
        verifyOrderConfirmation("Payment gateway integration seamless");
    }

    @Given("there are changes to the user's checkout status")
    public void thereAreChangesToTheUsersCheckoutStatus() {
        // Simulate status change
    }

    @When("a change occurs")
    public void aChangeOccurs() {
        // Change handling logic
    }

    @Then("the user should be notified of the changes")
    public void theUserShouldBeNotifiedOfTheChanges() {
        // Assume notification is verified
        verifyOrderConfirmation("User notified of changes");
    }

    @Given("the user proceeds through the checkout process")
    public void theUserProceedsThroughTheCheckoutProcess() {
        proceedToCheckout();
    }

    @When("they interact with all features")
    public void theyInteractWithAllFeatures() {
        enterShippingDetails("John Doe", "123 Main St", "Anytown", "12345", "555-1234");
        selectPaymentMethod("Credit Card");
        placeOrder();
    }

    @Then("each feature should function correctly and reliably")
    public void eachFeatureShouldFunctionCorrectlyAndReliably() {
        verifyOrderConfirmation("All features function correctly");
    }
}